### PR TITLE
Setup architecture

### PR DESCRIPTION
**Pull request by deandreamatias-worldline**
- Create project and implement minimal features
- Add documentation
- Create GitHub templates
- Add Github action for basic verification

### CODE REVIEW

File: ios/Podfile.lock
Comment (sergiocasero) [inline ios/Podfile.lock:1]: We should ignore .lock files as they're auto generated files

File: lib/di/di.config.dart
Comment (sergiocasero) [inline lib/di/di.config.dart:1]: As this is an autogenerated file, what's your opinion about ignoring it?

File: lib/env/constants.dart [hunk 1]
```diff
@@ -0,0 +1,3 @@
+class Constants {
+  static const String enlaceBack = 'https://worldline.com/es-es/home.html';
```
Comment (sergiocasero) [inline lib/env/constants.dart]: spanish word?? haha

**Review by sergiocasero — CHANGES_REQUESTED** [review]
Please, remember that :

/ios/Runner.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist
/ios/Runner.xcodeproj/project.xcworkspace/contents.xcworkspacedata
/ios/Runner.xcodeproj/xcshareddata

must be ignored :), nice job!

File: lib/di/di.config.dart
Comment (deandreamatias-worldline) [reply lib/di/di.config.dart:1]: Not ignore: add effort to generate the files on each project setup (by developer or pipeline).
Ignoring: When we has some generated files, is ok. But if the project grow and add packages like json_serializable or freezed, that would be a lot of files to change on commits and review on PR.

My opinion: I'm thing that would be nice ignoring. On almost all project that I worked, I ignored this files

File: lib/env/constants.dart (hunk 1 above)
Comment (deandreamatias-worldline) [reply lib/env/constants.dart]: Copy and paste. Sorry hehehe

**Comment by deandreamatias-worldline** [comment]
> Please, remember that :
> 
> /iosApp/Runner.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist /iosApp/Runner.xcodeproj/project.xcworkspace/contents.xcworkspacedata /iosApp/Runner.xcodeproj/xcshareddata
> 
> must be ignored :), nice job!

Inside /iosApp/Runner.xcodeproj/xcshareddata, has the xscheme files, like `dev.xcscheme`. I think that this folder is important to don't ignore

**Pull request merged**